addons/pos_hr/static/src/js/models.js

```diff
@@ -5,12 +5,18 @@ var models = require('point_of_sale.models');
 
 models.load_models([{
     model:  'res.users',
-    fields: ['name','groups_id'],
-    domain: function(self){ return [['company_id','=',self.user.company_id[0]],'|', ['groups_id','=', self.config.group_pos_manager_id[0]],['groups_id','=', self.config.group_pos_user_id[0]]]; },
+    fields: ['name','groups_id', 'company_ids'],
+    domain: function(self){ return ['|', ['groups_id','=', self.config.group_pos_manager_id[0]],['groups_id','=', self.config.group_pos_user_id[0]]]; },
```

Comment:
you can add `['company_ids', 'in', self.config.company_id[0] ]` ti the domain. This way you do not have to do the extra forEach and don't have to load the company_ids field.